subsys/net/l2/ppp/lcp.c

```diff
@@ -179,6 +179,10 @@ static void lcp_down(struct ppp_fsm *fsm)
 
 	ppp_link_down(ctx);
 
+	if (!ctx->is_net_carrier_up) {
```

Comment:
This flag does not actually seem to be set anywhere no more, so this check will always pass. Would it make sense to use `net_if_is_carrier_ok()` on the interface instead? `is_net_carrier_up` should probably be removed.